hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleUniqueToken.java

```diff
@@ -21,40 +21,73 @@
  */
 
 import com.google.common.base.MoreObjects;
+import com.hedera.services.state.merkle.internals.IdentityCodeUtils;
 import com.hedera.services.state.submerkle.EntityId;
 import com.hedera.services.state.submerkle.RichInstant;
 import com.swirlds.common.io.SerializableDataInputStream;
 import com.swirlds.common.io.SerializableDataOutputStream;
 import com.swirlds.common.merkle.utility.AbstractMerkleLeaf;
 
 import java.io.IOException;
+import java.time.Instant;
+import java.util.Arrays;
 import java.util.Objects;
 
 /**
  * Represents an uniqueToken entity. Part of the nft implementation.
  */
 public class MerkleUniqueToken extends AbstractMerkleLeaf {
-	public static final int UPPER_BOUND_METADATA_BYTES = 1024;
+	private static final int TREASURY_OWNER_CODE = 0;
+
 	static final int MERKLE_VERSION = 1;
 	static final long RUNTIME_CONSTRUCTABLE_ID = 0x899641dafcc39164L;
 
-	private EntityId owner;
-	private RichInstant creationTime;
+	public static final int UPPER_BOUND_METADATA_BYTES = 1024;
+
+	private int ownerCode;
+	private int creationNanos;
+	private long creationSecs;
 	private byte[] metadata;
 
 	/**
-	 * @param owner        The entity which owns the unique token.
-	 * @param metadata     Metadata about the token.
-	 * @param creationTime The consensus time at which the token was created.
+	 * Constructs a Merkle-usable unique token from an explicit owner id.
+	 *
+	 * @param owner
+	 * 		the id of the entity which owns the unique token
+	 * @param metadata
+	 * 		metadata about the token
+	 * @param creationTime
+	 * 		the consensus time at which the token was created
 	 */
 	public MerkleUniqueToken(
 			EntityId owner,
 			byte[] metadata,
 			RichInstant creationTime
 	) {
-		this.owner = owner;
+		this.ownerCode = owner.identityCode();
+		this.metadata = metadata;
+		this.creationSecs = creationTime.getSeconds();
+		this.creationNanos = creationTime.getNanos();
```

Comment:
just do a
```
return MerkleUniqueToken(owner.identityCode(), metadata, creationTime.getSeconds(), creationTime.getNanos()
```
to avoid code duplication